CMakeLists.txt

```
@@ -211,19 +228,26 @@ if (NOT (CMAKE_VERSION VERSION_LESS 3.13))
   cmake_policy(SET CMP0079 NEW)
 endif()
 
-add_subdirectory(vendor/xtl)
-set(xtl_DIR ${CMAKE_CURRENT_BINARY_DIR}/vendor/xtl)
-add_subdirectory(vendor/xtensor)
+if (NOT xtl_FOUND)
+  add_subdirectory(vendor/xtl)
+  set(xtl_DIR ${CMAKE_CURRENT_BINARY_DIR}/vendor/xtl)
+endif()
+
+if (NOT xtensor_FOUND)
+  add_subdirectory(vendor/xtensor)
+endif()
```

I think it's safe to assume that if xtensor is not found, we should use the submodules for both xtensor and xtl:
```suggestion
if (NOT xtensor_FOUND)
  add_subdirectory(vendor/xtl)
  set(xtl_DIR ${CMAKE_CURRENT_BINARY_DIR}/vendor/xtl)
  add_subdirectory(vendor/xtensor)
endif()
```